### ISSUE TITLE
service selector for jaeger-query service is incorrect

### ISSUE THREAD

**Issue by harshal-shah** [jaegertracing/jaeger-operator#170]
I'm using jaeger-operator version 1.8.2
When installing jaeger with agent mode as daemonset and following spec: 

~~~
apiVersion: "io.jaegertracing/v1alpha1"
kind: "Jaeger"
metadata:
  name: jaeger
  labels:
    app: jaeger
    chart: "jaeger-operator-0.1.0"
    release: "jaeger"
    heritage: "Tiller"
spec:
  strategy: allInOne
  allInOne:
    image: jaegertracing/all-in-one:1.8
    options:
      log-level: debug
      query:
        base-path: /
  storage:
    options:
      max-traces: 50000
      memory: null

  agent:
    strategy: DaemonSet
    options:
      log-level: debug
  ingress:
    enabled: false
~~~

Pods are generated with following labels: 

~~~
NAME                               READY   STATUS    RESTARTS   AGE   LABELS
jaeger-547b6fc6cd-cj7vx            1/1     Running   0          3m    app=jaeger,jaeger=jaeger,pod-template-hash=1036297278
jaeger-agent-daemonset-6b77g       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-7lkfv       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-8s7m7       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-9tr46       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-crrch       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-flcx4       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-g8s49       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-gtltt       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
jaeger-agent-daemonset-n5pqn       1/1     Running   0          3m    app=jaeger,controller-revision-hash=424870228,jaeger-component=agent-daemonset,jaeger=jaeger,pod-template-generation=1
~~~

And `jaeger-query` service is created with the following selector: 
~~~
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2019-01-03T14:36:05Z"
  labels:
    app: jaeger
    jaeger: jaeger
  name: jaeger-query
  namespace: tracing
  ownerReferences:
  - apiVersion: io.jaegertracing/v1alpha1
    controller: true
    kind: Jaeger
    name: jaeger
    uid: e6fe3819-0f64-11e9-991a-0a2f2f061794
  resourceVersion: "120613599"
  selfLink: /api/v1/namespaces/tracing/services/jaeger-query
  uid: e708f3fc-0f64-11e9-991a-0a2f2f061794
spec:
  clusterIP: None
  ports:
  - name: query
    port: 16686
    protocol: TCP
    targetPort: 16686
  selector:
    app: jaeger
    jaeger: jaeger
  sessionAffinity: None
  type: ClusterIP
~~~

This results in jaeger-query pointing to all agent pods as well instead of pointing to just the pod created by jaeger deployment. This seems incorrect and the jaeger deployment should create a specific label for pods created by `jaeger` deployment which the service could use in its selector 
something like `component: jaeger-query`

**Comment by objectiser** [comment]
Thanks for reporting this - we have recently been discussing whether we should change the labelling approach, so will factor this issue in.

**Comment by jpkrohling** [comment]
This should have been fixed by #172. If it doesn't solve your problem, feel free to reopen this issue.
